third-party/rsutils/include/rsutils/easylogging/easyloggingpp.h

```diff
@@ -41,17 +40,63 @@
-#define LOG_WARNING(...) do { CLOG(WARNING , LIBREALSENSE_ELPP_ID) << __VA_ARGS__; } while(false)
-#define LOG_ERROR(...)   do { CLOG(ERROR   , LIBREALSENSE_ELPP_ID) << __VA_ARGS__; } while(false)
-#define LOG_FATAL(...)   do { CLOG(FATAL   , LIBREALSENSE_ELPP_ID) << __VA_ARGS__; } while(false)
+// Direct log to ELPP, without conversion to string first
+// We've seen cases where this fails in U22, causing weird effects with custom overloads/types (e.g., json)
+#define LIBRS_LOG_STR_( LEVEL, STR )                                                                                   \
+    do                                                                                                                 \
+    {                                                                                                                  \
+        auto logger__ = el::Loggers::getLogger( rsutils::g_librealsense_elpp_id );                                     \
+        if( logger__ && logger__->enabled( el::Level::LEVEL ) )                                                        \
+        {                                                                                                              \
+            el::base::Writer( el::Level::LEVEL, __FILE__, __LINE__, ELPP_FUNC, el::base::DispatchAction::NormalLog )   \
+                    .construct( logger__ )                                                                             \
+                << STR;                                                                                                \
+        }                                                                                                              \
+    }                                                                                                                  \
+    while( false )
+
+// Same, but first convert to string using usual mechanisms
+#define LIBRS_LOG_( LEVEL, ... )                                                                                       \
+    do                                                                                                                 \
+    {                                                                                                                  \
+        auto logger__ = el::Loggers::getLogger( rsutils::g_librealsense_elpp_id );                                     \
+        if( logger__ && logger__->enabled( el::Level::LEVEL ) )                                                        \
+        {                                                                                                              \
+            std::ostringstream os__;                                                                                   \
+            os__ << __VA_ARGS__;                                                                                       \
+            el::base::Writer( el::Level::LEVEL, __FILE__, __LINE__, ELPP_FUNC, el::base::DispatchAction::NormalLog )   \
```

Comment:
IS this the same code as implemented in `CLOG`?

Reply:
Yes, except the construct() call is simpler and more efficient.